explainer.md

```diff
@@ -186,225 +181,236 @@ https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b
-  animation-delay: 300ms;
+::page-transition {
+  position: absolute;
+  top: 0;
+  left: 0;
+}
+
+::page-transition::image-wrapper {
+  position: absolute;
+  inset: 0;
+  isolation: isolate;
+}
+
+::page-transition::image {
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  /* This is a new proposed value. See 'cross-fading' */
+  mix-blend-mode: lighter;
 }
 ```
 
-CSS can be used to make changes to the default animation, or override `animation-name` to remove the default.
+#### Transition animations
 
-### JavaScript access to elements
+The default animations are as follows:
 
-So far the transition elements have been addressed by pseudo-element selectors, but JavaScript could be given access to the elements using the [CSSPseudoElement](https://drafts.csswg.org/css-pseudo-4/#CSSPseudoElement-interface) interface.
+- The `::image(incoming)` animates _from_ opacity 0 with a fill-mode of `both`.
+- The `::image(outgoing)` animates _to_ opacity 0 with a fill-mode of `both`.
+- If an element exists in both Page-A and Page-B, a transform animation takes its container from Page-A size/transform to Page-B size/transform.
 
-```js
-// In Page-B
-document.performTransition((transition) => {
-  // Build up transition parts, then…
-  document.documentElement.pseudo("page-transition-container(name)").animate(...);
-});
-```
+The styles to apply these animations will be dynamically added to the UA stylesheet, and removed once the transition is complete.
+
+- Open question: What are the default durations and easings?
+- Open question: Are the generated keyframes usable outside the UA stylesheet? If so, what are they called?
+- Open question: I worry that the default scaling animation makes it hard to do animations where both images cross-fade but don't change size, which feels like a common type of transition. I think this needs more thought.
+- [Open question](https://github.com/WICG/shared-element-transitions/issues/84): Default animations work well for things which are at least partially in-viewport in both Page-A and Page-B, but it gets tricky if you consider a non-sticky header that scrolled out of view by 1000s of pixels.
+- [Open question](https://github.com/WICG/shared-element-transitions/issues/83): When will the default animation start? When the browser would usually first render Page-B?
+
+All of these can be overridden by the developer via CSS animations.
+
+The animation is considered complete once all animations on all the pseudo-elements finish.
+
+### Via JavaScript
+
+The JavaScript API extends on the capabilities of the CSS API.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/86) What's the deadline for calling `performTransition`?
+```js
+document.addEventListener("beforepageshow", async (event) => {
+  if (!event.transition) return;
+  event.transition.setElement(document.documentElement, "root");
+  event.transition.setElement(document.querySelector(".header"), "header");
```

Comment:
I'm not sure if discuss this below, but one problem of this is that if we want to wait for `.header` to at least be parsed before `beforepageshow` fires, but it's unclear if UA can know what are the important elements, since this is only revealed as a part of `beforepageshow` handler

Reply:
Yeah, eventually we'll need a way to wait for stuff. Question is if we need it for v1.

Reply:
That problem is going to be an issue for the purely declarative stuff too, we wouldn't know if we've fetched/parsed enough of the document for all shared elements to be discovered. It's better to tackle this as a separate problem (like render-blocking) such that it can be used with transitions later. Until then, we can rely on UA heuristics for this.